ecs-init/docker/docker.go

```diff
@@ -571,6 +576,25 @@ func nvidiaGPUDevicesPresent() bool {
 	return true
 }
 
+// nvidiaGPUDevicesPresentWithRetries checks if NVIDIA GPU devices are present in the instance. It retries if NVIDIA
+// GPU devices are not yet present every `nvidiaGPUDevicesPresentRetryTime` interval of time up to a maximum of
+// `nvidiaGPUDevicesPresentMaxRetries` retries.
+func nvidiaGPUDevicesPresentWithRetries() bool {
+	nvidiaGPUDevicesPresentRetryTime := 3 * time.Second
```

Comment:
super nit(non-blocking): could these be defined as constants?

Reply:
I was thinking we should keep them defined within the function since I don't think they should be used outside of the context of this function. Is there an advantage to having them defined as constants outside of the function that I might be missing?

Reply:
defining these as file-level constants would help us track all constant values in a single place and increase readability.

there's a bunch of them already here: https://github.com/aws/amazon-ecs-agent/blob/cf8c7a6b65043c550533f330b10aef6d0a342214/ecs-init/docker/docker.go#L35

Reply:
I think all the constants linked here have use cases to be reused across multiple places within the  `docker` package, whereas the values here only have a single place where it would be applicable to use them.

Reply:
+1 It'll be a bit easier/more organized to keep all of the constants defined in one place IMO in the case we need to change it for future references (but I don't have a too strong of an opinion here).

Reply:
Discussed offline - overall aligned that we will move these to the constants defined at the top of the file for enhanced readability.